test/audit/AuditServiceSpec.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import play.api.test.FakeRequest
+import play.api.test.Helpers.running
+import uk.gov.hmrc.auth.core.{AffinityGroup, User}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+
+import base.SpecBase
+import models.WhatIsYourRoleAsImporter.{AgentOnBehalfOfOrg, EmployeeOfOrg}
+import models.events.{AgentIndicatorEvent, UserTypeEvent}
+import models.requests.{DataRequest, OptionalDataRequest}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchersSugar.eqTo
+import org.mockito.Mockito
+import org.mockito.Mockito.doNothing
+import org.mockito.MockitoSugar.mock
+import org.scalatest.prop.TableDrivenPropertyChecks
+
+class AuditServiceSpec extends SpecBase with TableDrivenPropertyChecks {
+
+  private val mockAuditConnector = mock[AuditConnector]
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    Mockito.reset(mockAuditConnector)
+  }
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  "sendUserTypeEvent" - {
+
+    implicit val optionalDataRequest: OptionalDataRequest[_] = OptionalDataRequest(
+      request = FakeRequest(),
+      userId = userAnswersId,
+      eoriNumber = EoriNumber,
+      affinityGroup = AffinityGroup.Individual,
+      credentialRole = Option(User),
+      userAnswers = None
+    )
+
+    val event = UserTypeEvent(
+      optionalDataRequest.userId,
+      optionalDataRequest.eoriNumber,
+      optionalDataRequest.affinityGroup,
+      optionalDataRequest.credentialRole
+    )
+
+    "must send userTypeEvent to auditConnector" in {
+      doNothing()
+        .when(mockAuditConnector)
+        .sendExplicitAudit(eqTo("UserEntersService"), eqTo(event))(any(), any(), any())
+
+      running(applicationBuilder().build()) {
```

Comment:
The built application isn't used at all here, and `running` should never be required anymore as it's to do with setting global state that we don't use in our apps.